model/app.model.js

```diff
@@ -7,15 +7,64 @@ const selectTopics = () => {
     });
 };
 
-const selectArticle = (id) => {
+const selectArticleById = (id) => {
     const query = "SELECT * FROM articles WHERE article_id = $1;";
     let params = [id];
     return db.query(query, params).then((result) => {
         return result.rows;
     });
 };
 
+const selectArticles = () => {
+    return db
+        .query(
+            `SELECT 
+            articles.article_id,
+            articles.author,
+            articles.title,
+            articles.topic,
+            articles.created_at,
+            articles.votes,
+            articles.article_img_url,
+            COALESCE(SUM(comments.comment_id),0) as comments_count
+    FROM articles LEFT JOIN comments
+    ON articles.article_id = comments.article_id
+    GROUP BY  articles.article_id, articles.author, articles.title, articles.topic, articles.created_at,   articles.votes, articles.article_img_url
+    ORDER BY articles.created_at DESC;`
+        )
+        .then((result) => {
+            return result.rows;
+        });
+};
+
+const selectAllCommentsForArticle = (id) => {
+    const query =
```

Comment:
In order to check whether the ID is valid we can call our earlier model where we get the article by it's ID. This model should already throw an error if the ID does not exist (assuming you added the error handling in the model and not the controller) 

If we pass this and continue to get an empty array in our .then, we know the issue isn't the ID but instead that there is no comments. Therefore we are just return the empty array. Make sure to also add a test for this and submit when done. If stuck, please send an NC help.